cpp/memilio/compartments/simulation.h

```diff
@@ -146,6 +146,160 @@ class Simulation
     OdeIntegrator m_integrator;
 }; // namespace mio
 
+/**
+ * @brief A class for the simulation of a compartment model with Flows. The Flows are integrated when calling the time integration scheme.
+ * @tparam M a CompartmentModel type
+ */
+template <class M>
+class FlowSimulation
+{
+    static_assert(is_compartment_model<M>::value, "Template parameter must be a compartment model.");
+
+public:
+    using Model = M;
```

Comment:
We need the scoped type Model (i.e. `public: using Model;`) so that we can get the Model type from a simulation, e.g.
```
template <class Sim>
void foo(Sim sim) {
    Sim::Model model();
    ...
}
```
But only `template <class Model> class Simulation { ... using Model; ... };` does not make sense, we need a "qualified type" to assign to Model, hence the template param (with a different name) M.